src/services/findAllReferences.ts

```diff
@@ -2004,20 +2023,51 @@ namespace ts.FindAllReferences {
 
         function getReferencesForStringLiteral(node: StringLiteralLike, sourceFiles: readonly SourceFile[], checker: TypeChecker, cancellationToken: CancellationToken): SymbolAndEntries[] {
             const type = getContextualTypeFromParentOrAncestorTypeNode(node, checker);
+            const callExpr = findAncestor(node, isCallExpression);
             const references = flatMap(sourceFiles, sourceFile => {
                 cancellationToken.throwIfCancellationRequested();
                 return mapDefined(getPossibleSymbolReferenceNodes(sourceFile, node.text), ref => {
-                    if (isStringLiteralLike(ref) && ref.text === node.text) {
-                        if (type) {
-                            const refType = getContextualTypeFromParentOrAncestorTypeNode(ref, checker);
-                            if (type !== checker.getStringType() && type === refType) {
-                                return nodeEntry(ref, EntryKind.StringLiteral);
-                            }
-                        }
-                        else {
+                    // Special case: Every string literal refers at least to itself.
+                    if (node === ref) return nodeEntry(ref, EntryKind.StringLiteral);
+
+                    // When evaluating references for a string literal, guarantee that the string literal
+                    // node is either global or that any reference shares the same scope.  Global literals
+                    // might all refer to each other, but if the string literal being considered is in a
+                    // call expression, any reference should also be from the same call expression.
+                    const refHasSameScope = !callExpr || callExpr === findAncestor(ref, isCallExpression);
+                    const refHasSameText = isStringLiteralLike(ref) || isPropertyNameLiteral(ref) && getTextOfIdentifierOrLiteral(ref) === node.text;

```

Comment:
Currently all string literal `ref`s are true without checking whether their text is `=== node.text`. I think it should instead be:

```suggestion
                    const refHasSameText = (isStringLiteralLike(ref) || isPropertyNameLiteral(ref)) && getTextOfIdentifierOrLiteral(ref) === node.text;
```